drivers/SmartThings/philips-hue/src/hue/api.lua

```diff
@@ -86,7 +86,7 @@ local function process_rest_response(response, err, partial, err_callback)
       )
     end
 
-    return table.unpack(json_result)
+    return table.unpack(json_result, 1, json_result.n)
```

Comment:
would you mind enlightening me about this call? I wasn't even aware `table.unpack` took multiple arguments

Reply:
It turns out that if you don't pass in a start and end point, `table.unpack` uses `#list` to determine the endpoint instead of using the `n` key that `table.pack` creates… which kinda defeats the point of using `table.pack` because you'll end up stopping at the first `nil` 😵‍💫

Reply:
See also: https://www.lua.org/manual/5.3/manual.html#pdf-table.unpack

Reply:
wouldn't it be better to only return non-`nil` values? or is it useful to keep them around?

Reply:
Yes, but, returning `nil, "error message string"` is a Lua convention for communicating failure, so we don't always control if the first argument is nil or not in the error case.